pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/RecordTransformer.java

```diff
@@ -43,4 +44,15 @@ default boolean isNoOp() {
    */
   @Nullable
   GenericRow transform(GenericRow record);
+
+  /**
+   * Transforms a record based on some custom rules using record reader context.
+   * @param record Record to transform
+   * @return Transformed record, or {@code null} if the record does not follow certain rules.
+   */
+  @Nullable
+  default GenericRow transformUsingRecordReaderContext(GenericRow record,
```

Comment:
just call it `transform(record, config)`, as it's easy to tell them apart by param list.

I'd suggest to pass in RecordReaderConfig (the interface), and make RecordReaderFileConfig implement RecordReaderConfig (which is an empty interface right now)

Reply:
Yes this makes sense to me.